Lib/glyphsLib/builder/anchors.py

```diff
@@ -125,3 +160,37 @@ def to_glyphs_glyph_anchors(self, ufo_glyph, layer):
         anchor.name = ufo_anchor.name
         anchor.position = Point(ufo_anchor.x, ufo_anchor.y)
         layer.anchors.append(anchor)
+
+
+def _component_closest_to_origin(components, glyph_set):
+    """Return the component whose (xmin, ymin) bounds are closest to origin.
+
+    This ensures that a component that is moved below another is
+    actually recognized as such. Looking only at the transformation
+    offset can be misleading.
+    """
+    return min(components, key=lambda comp: _distance((0, 0), _bounds(comp, glyph_set)))
+
+
+def _distance(pos1, pos2):
+    x1, y1 = pos1
+    x2, y2 = pos2
+    return (x1 - x2) ** 2 + (y1 - y2) ** 2
+
+
+def _is_ligature_mark(glyph):
+    return not glyph.name.startswith("_") and "_" in glyph.name
+
+
+def _bounds(component, glyph_set):
+    """Return the (xmin, ymin) of the bounds of `component`."""
+    if hasattr(component, "bounds"):  # e.g. defcon
+        return component.bounds[:2]
+    elif hasattr(component, "draw"):  # e.g. ufoLib2
+        pen = fontTools.pens.boundsPen.BoundsPen(glyphSet=glyph_set)
```

Comment:
ufoLib2 has Component.getBounds(layer) method, since v0.6.1:
https://github.com/fonttools/ufoLib2/blob/e9a5c5d2d1a0ddd66e6e8f51e7c58f2e494f164b/src/ufoLib2/objects/component.py#L42-L49

Reply:
I can change that if you like, but it means introducing `ufoLib2` as a dependency. `fontTools` is already a dependency today, so that works. I don't know which is better here. Let me know.

Reply:
ufoLib2 is already a dependency of glyphsLib, see https://github.com/googlefonts/glyphsLib/blob/1ad88c1dedb3072f1d9e62dc58b268496a1c6bc3/setup.cfg#L37

Reply:
Okay. I overlooked that, sorry

Reply:
I don't understand how `getBounds()` works, especially that thing with the Drawable.
However, the method of bounds calculation isn't related to my issue here with the anchors, so I will skip this one and leave it to you. It works without changes.
I’m going into a summer break in two days and need to finish this up. Sorry for that.

Reply:
no problem!

basically you would just do `return return component.getBounds(glyph_set)`